src/DI/ElasticsearchExtension.php

```diff
@@ -5,25 +5,40 @@
 use Elasticsearch\Client;
 use Elasticsearch\ClientBuilder;
 use Nette\DI\CompilerExtension;
+use Nette\Schema\Expect;
+use Nette\Schema\Schema;
+use stdClass;
 
+/**
+ * @property-read stdClass $config
+ */
 class ElasticsearchExtension extends CompilerExtension
 {
 
-	/** @var mixed[] */
-	private $defaults = [
-		'hosts' => ['localhost'],
-	];
+	public function getConfigSchema(): Schema
+	{
+		return Expect::structure([
+			'hosts' => Expect::anyOf(Expect::arrayOf('string'), Expect::structure([
+				'host' => Expect::string()->required(),
+				'port' => Expect::int(),
+				'schema' => Expect::string(),
+				'path' => Expect::string(),
+				'user' => Expect::string(),
+				'pass' => Expect::string(),
+			])),
```

Comment:
Is it possible to use
```yaml
hosts
  - localhost
  - host: 192.168.0.1
    port: 1234
```
or could it be just array of strings or array of arrays?

Reply:
Elasticsearch-php itself supports it and updated test are passing to.

```php
<?php declare(strict_types = 1);

use Contributte\Elasticsearch\DI\ElasticsearchExtension;
use Elasticsearch\Client;
use Nette\DI\Compiler;
use Nette\DI\Container;
use Nette\DI\ContainerLoader;
use Tester\Assert;
use Tester\FileMock;

test(function (): void {
	$loader = new ContainerLoader(TEMP_DIR, true);
	$class = $loader->load(function (Compiler $compiler): void {
		$compiler->addExtension('elasticsearch', new ElasticsearchExtension());
		$compiler->loadConfig(FileMock::create('
			elasticsearch:
					hosts:
					    - localhost
					    -
					    	host: 192.168.1.100
					    	port: 9999
		', 'neon'));
	}, '1a');

	/** @var Container $container */
	$container = new $class();

	Assert::type(Container::class, $container);
	Assert::type(Client::class, $container->getService('elasticsearch.client'));
});
```

Reply:
I was not sure if nette/schema would allow it. 
I am okay with PR then 👍

Reply:
I was not sure either, but I played with the Schema for a while and got it working as intended, no BC breaks for configuration then.

I am gonna push the 2nd test and then merge this PR.